src/components/map/mapbox/controls/DrawControl.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Polygon } from "geojson";
+import * as turf from "@turf/turf";
+import { Map } from "mapbox-gl";
+import MapboxDraw from "@mapbox/mapbox-gl-draw";
+
+import "mapbox-gl/dist/mapbox-gl.css";
+import "@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.css";
+import DrawRectangleControl from "./DrawRectangleControl";
+import {
+  BBoxCondition,
+  DownloadConditionType,
+  IDownloadCondition,
+} from "../../../../pages/detail-page/context/DownloadDefinitions";
+
+interface DrawControlProps {
+  map: Map | undefined | null;
+  setDownloadConditions: (
+    type: DownloadConditionType,
+    conditions: IDownloadCondition[]
+  ) => void;
+  draw: MapboxDraw;
+}
+
+const DrawControl: React.FC<DrawControlProps> = ({
+  map,
+  setDownloadConditions,
+  draw,
+}) => {
+  const [open, setOpen] = useState<boolean>(false);
+
+  const anchorRef = useRef(null);
+  const popperRef = useRef<HTMLDivElement>(null);
+
+  const updateArea = useCallback(() => {
```

Comment:
Can you put the updateArea inside the on off event block so that it didn't need to use callback?